bson/bsoncodec/struct_codec.go

```diff
@@ -98,6 +98,21 @@ func (sc *StructCodec) EncodeValue(r EncodeContext, vw bsonrw.ValueWriter, val r
 
 		desc.encoder, rv, err = defaultValueEncoders.lookupElementEncoder(r, desc.encoder, rv)
```

Comment:
We should have a similar error check here for non-nil and not `errInvalidValue` like we do above:
```
if lookupErr != nil && lookupErr != errInvalidValue {
    return lookupErr
}
```

Reply:
done